### PR TITLE
Use refactored `curl_download` method.

### PR DESCRIPTION
**Pull request by reitermarkus**

### CODE REVIEW

File: cmd/brew-pull-linux.rb [hunk 1]
```diff
@@ -298,7 +298,7 @@ def fetch_patch
       extra_msg = @description ? "(#{@description})" : nil
       ohai "Fetching patch #{extra_msg}"
       puts "Patch: #{patch_url}"
-      curl patch_url, "-s", "-o", patchpath
+      curl_download patch_url, "-s", '--continue-at', '0', to: patchpath
```
Comment (ilovezfs) [inline cmd/brew-pull-linux.rb]: what's the reason for single quotes?

**Comment by sjackman** [comment]
Thanks, Markus. Do you know why the `curl` method started failing? Just curious.

File: cmd/brew-pull-linux.rb (hunk 1 above)
Comment (reitermarkus) [inline cmd/brew-pull-linux.rb]: None, other than being tired. 😄

**Comment by sjackman** [comment]
Once the single quotes are changed to double quotes, and it passes CI, I'm happy to merge this PR.

File: cmd/brew-pull-linux.rb (hunk 1 above)
Comment (ilovezfs) [inline cmd/brew-pull-linux.rb]: :zzz:

**Comment by reitermarkus** [comment]
Yes, by default now, `curl_download` uses `--continue-at -`, so a download will continue instead of restart. In this case though, the server doesn't support this:

```
curl: (33) HTTP server doesn't seem to support byte ranges. Cannot resume.
```

**Comment by reitermarkus** [comment]
This should be merged first: https://github.com/Homebrew/brew/pull/3030

**Comment by sjackman** [comment]
I feel that `curl_download` should silently fall back to starting at 0 if the server doesn't support resume.

**Comment by reitermarkus** [comment]
Yes, might be a good idea to `retry` once if `curl` errors with `33`.

**Comment by sjackman** [comment]
Blocked by https://github.com/Homebrew/brew/pull/3030

**Comment by sjackman** [comment]
Thanks, Markus.
Homebrew/brew#3030 is merged.
This PR is now blocked by merging the yet unreleased Homebrew/brew 1.3.2 into Linuxbrew/brew

**Comment by maxim-belkin** [comment]
> This PR is now blocked on merging the yet unreleased Homebrew/brew 1.3.2 into Linuxbrew/brew.

hwy?

**Comment by sjackman** [comment]
The `curl_download` function that it uses is not yet in Linuxbrew/brew.
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/utils/curl.rb#L40
https://github.com/Linuxbrew/brew/blob/master/Library/Homebrew/utils/curl.rb#L34

**Comment by sjackman** [comment]
@jonchang 's PR https://github.com/Linuxbrew/homebrew-developer/pull/65 should resolve this issue. Once `Homebrew/brew` 1.3.2 is released and merged into `Linuxbrew/brew` we can merge this PR.

**Comment by iMichka** [comment]
What is the stats on this?

**Comment by sjackman** [comment]
Merged. Thanks for the reminder, Michka.

**Comment by iMichka** [comment]
Cool. Thanks.

**Pull request merged**